toolsrc/src/vcpkg/binarycaching.cpp

```diff
@@ -663,6 +663,21 @@ namespace
     const ExpectedS<fs::path>& default_cache_path()
     {
         static auto cachepath = System::get_platform_cache_home().then([](fs::path p) -> ExpectedS<fs::path> {
+            auto maybe_cachepath = System::get_environment_variable("VCPKG_BINARY_CACHE");
+            if (auto p_str = maybe_cachepath.get())
+            {
+                const auto path = fs::u8path(*p_str); //(u8path is deprecated in C++20)
```

Comment:
```suggestion
                const auto path = fs::u8path(*p_str);
```

`u8path` is deprecated in 20, but we won't be able to change over any time soon to `u8string`, and as of right now it's unimportant to change over. At the point where we can reasonably expect `char8_t` on all compilers, we'll actually start switching over.